tests/providers/node/pool_test.py

```diff
@@ -0,0 +1,47 @@
+import pytest
+
+from ethtx.providers.node.base import NodeBase
+from ethtx.providers.node.pool import NodeConnectionPool
+
+MAINNET_CHAIN = {"mainnet": {"hook": "a", "poa": True}}
+GOERLI_CHAIN = {"goerli": {"hook": "a, b, c", "poa": False}}
+
+GOERLI_NODE = NodeBase("B", False)
+
+
+class TestNodeConnectionPool:
+    @classmethod
+    def setup_class(cls):
+        cls.pool = NodeConnectionPool(nodes=MAINNET_CHAIN)
```

Comment:
It would be better to create new pool for every test, just to make sure starting state for tests is always the same.